lib/house.rb

```diff
@@ -0,0 +1,62 @@
+
+class House
+
+  attr_reader :address, :price, :rooms, :details, :rooms_by_category
+
+  def initialize(price_parameter, address_parameter)
+    @price = price_parameter.delete_prefix('$').to_i
+    @address = address_parameter
+    @rooms = []
+    @details = {"price" => @price, "address" => @address}
+    @rooms_by_category = {}
```

Comment:
Similarly, we don't need to store rooms by category, because a room stores its own category. This would be better as a separate method.